regulation-worker/internal/delete/batch/s3.go

```diff
@@ -0,0 +1,58 @@
+package batch
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/rudderlabs/rudder-server/regulation-worker/internal/model"
+)
+
+type S3DeleteManager struct {
+}
+
+//reason behind using sed: https://www.rtuin.nl/2012/01/fast-search-and-replace-in-large-files-with-sed/
+//Delete user details corresponding to `userAttributes` from `uncompressedFileName` & delete `uncompuressedFileName`
+func (dm *S3DeleteManager) delete(ctx context.Context, userAttributes []model.UserAttribute, uncompressedFileName string) ([]byte, error) {
+
+	//actual delete
+	var n int
+	for i := 0; i < len(userAttributes); i++ {
+		n += len(userAttributes[i].UserID) + 4
+		if userAttributes[i].Email != nil {
+			n += len(*userAttributes[i].Email) + 4
+		}
+		if userAttributes[i].Phone != nil {
+			n += len(*userAttributes[i].Phone) + 4
+		}
+	}
+
+	searchObject := make([]byte, 0, n)
+	for _, users := range userAttributes {
+		searchObject = append(searchObject, "/"...)
+		searchObject = append(searchObject, users.UserID...)
+		searchObject = append(searchObject, "/d;"...)
+
+		if users.Email != nil {
+			searchObject = append(searchObject, "/"...)
+			searchObject = append(searchObject, []byte(*users.Email)...)
+			searchObject = append(searchObject, "/d;"...)
+		}
+
+		if users.Phone != nil {
+			searchObject = append(searchObject, "/"...)
+			searchObject = append(searchObject, []byte(*users.Phone)...)
+			searchObject = append(searchObject, "/d;"...)
+		}
+	}
+
+	out, err := exec.Command("sed", "-e", string(searchObject), uncompressedFileName).Output()
```

Comment:
As we discussed there is a limit on the number of bytes you can provide as a single argument. In most cases is 4k bytes. I believe this is too restrictive for our use case. 

It better to write all the rules into a temporary file and provide this file instead:

you can use a [bufio writer](https://pkg.go.dev/bufio) to efficiently write to the file instead of appending all the rules in memory.

```suggestion
       out, err := exec.Command("sed", "-f", sedCommandsFile uncompressedFileName).Output()
```